File: src_python/modeles/resultat_traitement.py
from modeles.resultat import Resultat

class ResultatTraitement:
    def __init__(self, resultat: Resultat):
        self.resultat = resultat
        self.messages = []

    @property
    def resultat(self):
        return self._resultat
    @resultat.setter
    def resultat(self, value: Resultat):
        self._resultat = value


    def ajouter_message(self, message: str):
        self.messages.append(message)

File: TP2-eneonce-update/src/main/java/ca/uqam/mgl7460/tp1/types/modeles/ResultatTraitement.java
package ca.uqam.mgl7460.tp1.types.modeles;

import java.util.Iterator;

/**
 * object qui contient le resultat du traitement.
 * 
 * En plus du résultat en tant que tel, le traitement
 * d'une demande peut "collecter" plusieurs messages indiquant
 * ce va ou ne va pas avec une demande de prêt
 */
public interface ResultatTraitement {

    /**
     * retourne le resultat (une valeur parmi NONDETERMINE, ACCEPTEE, REFUSEE)
     * @return
     */
    public Resultat getResultat();

    /**
     * modifie le résultat
     * @param resultat
     */
    public void setResultat(Resultat resultat);

    /**
     * Retourne la liste de messages 'collectés' durant le traitement
     * @return
     */
    public Iterator<String> getMessages();

    /**
     * ajoute un message à la liste de message
     * @param message
     */
    public void ajouteMessage(String message);

}
